specs/xaml-resource-references-tracing-spec.md

```diff
@@ -0,0 +1,165 @@
+Tracing XAML resource reference lookup failures
+===
+
+# Background
+
+Failure to resolve a XAML resource reference (`{StaticResource}` or `{ThemeResource}`) is one of 
+the most common causes of app crashes. Such failures manifest as a stowed exception with the 
+message "Cannot find a Resource with the Name/Key *foo*", and generally fall into one of two 
+buckets:
+
+1. The referenced resource legitimately does not exist, e.g. the referenced key is misspelled or 
+the intended matching resource has not been added to the app.
+2. The referenced resource *does* exist in the app, but it is not reachable from the reference 
+at run-time.
+
+Unfortunately, the stowed exception message is the *only* information provided about the error, 
+which makes it difficult or, more often, outright impossible to debug. This spec describes a new 
+API on the `DebugSettings` class that developers can use to access more detailed information about 
+the resource reference lookup failure.
+
+# API Pages
+
+_(Each of the following L2 sections correspond to a page that will be on docs.microsoft.com)_
+
+## DebugSettings.XamlResourceReferenceFailed Event
+
+Occurs when a [XAML resource reference](https://learn.microsoft.com/en-us/windows/apps/design/style/xaml-resource-dictionary) cannot be resolved.
+
+_Spec note: is there a version of the 'ResourceDictionary and XAML resource references' article that 
+links to the Windows App SDK documentation for relevant APIs?_
+
+```c#
+public event TypedEventHandler<DebugSettings,XamlResourceReferenceFailedEventArgs> 
+XamlResourceReferenceFailed
+
+```
+
+### Remarks
+
+`IsXamlResourceReferenceTracingEnabled` must be `true` and there must be a debugger attached to 
+the app process in order for `XamlResourceReferenceFailed` to fire and for tracing to appear in 
+debugger output. You don't need to handle the event in order to see tracing appear in a debugger. 
+The debugger output contains message information that goes to the **Output** window of the 
+debugger. Attaching a `XamlResourceReferenceFailed` handler yourself is an advanced scenario for 
+when you want to see the raw message.
```

Comment:
This section talks about debug output but doesn't describe that it must be the native debugger. It would help readability to move that description from the args to here.